File: bixby/src/main/java/guepardoapps/lucahome/bixby/classes/actions/WirelessSwitchAction.java
package guepardoapps.lucahome.bixby.classes.actions;

import android.support.annotation.NonNull;

import java.util.Locale;

@SuppressWarnings("WeakerAccess")
public class WirelessSwitchAction implements IBixbyAction {
    private static final String Tag = WirelessSwitchAction.class.getSimpleName();

    private String _wirelessSwitchName;

    public WirelessSwitchAction(@NonNull String wirelessSwitchName) {
        _wirelessSwitchName = wirelessSwitchName;
    }

    public WirelessSwitchAction() {
        this("");
    }

    public String GetWirelessSwitchName() {
        return _wirelessSwitchName;
    }

    @Override
    public String GetDatabaseString() {
        return String.format(Locale.getDefault(), "%s", _wirelessSwitchName);
    }

    @Override
    public String GetInformationString() {
        return String.format(Locale.getDefault(), "%s \nfor %s -> Toggle", Tag, _wirelessSwitchName);
    }

    @Override
    public String toString() {
        return String.format(Locale.getDefault(),
                "{\"Class\":\"%s\",\"WirelessSocketName\":\"%s\"}",
                Tag, _wirelessSwitchName);
    }
}

File: bixby/src/main/java/guepardoapps/lucahome/bixby/models/actions/WirelessSwitchAction.kt
package guepardoapps.lucahome.bixby.models.actions

import android.support.annotation.NonNull
import guepardoapps.lucahome.bixby.models.shared.IBixbyEntity
import guepardoapps.lucahome.common.utils.Logger
import java.util.*

class WirelessSwitchAction(
        @NonNull var uuid: UUID = UUID.randomUUID(),
        @NonNull var name: String = "") : IBixbyEntity {
    private val tag = WirelessSwitchAction::class.java.simpleName

    override fun getDatabaseString(): String {
        return "$uuid:$name"
    }

    override fun getInformationString(): String {
        return "$tag: $name"
    }

    override fun parseFromDb(databaseString: String) {
        val data: Array<String> = databaseString.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (data.size == 2) {
            try {
                uuid = UUID.fromString(data[0])
                name = data[1]

            } catch (exception: Exception) {
                Logger.instance.error(tag, exception.toString())
                uuid = UUID.randomUUID()
                name = ""
            }

        } else {
            Logger.instance.error(tag, "Invalid data size ${data.size}")
            uuid = UUID.randomUUID()
            name = ""
        }
    }

    override fun toString(): String {
        return "{\"Class\":\"$tag\",\"Uuid\":\"$uuid\",\"Name\":\"$name\"}"
    }
}